lib/logstash/codecs/identity_map_codec.rb

```diff
@@ -0,0 +1,176 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "concurrent"
+
+# This class is a Codec duck type
+# Using Composition, it maps from a stream identity to
+# a cloned codec instance via the same API as a Codec
+
+module LogStash module Codecs class IdentityMapCodec
+  # subclass of Exception, LS has more than limit (100) active streams
+  class IdentityMapUpperLimitException < Exception; end
```

Comment:
ok, I see. but still, generally speaking I think a better strategy is to have a base error class and have all  class specific errors inherit from this base class so that its possible to have a specific class-wide rescue clause. in this case I guess it's debatable if we should consider this a hard or fatal error or an error that could be expected and rescued (for logging)

```ruby
class MyClassBaseError < StandardError; end
class FooError < MyClassBaseError; end

...
rescue MyClassBaseError => e
  # do something specific with all errors from MyClass
end
...
```